modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -357,8 +357,8 @@ public static BigDecimal toBigDecimal(Number value, int precision, int scale) {
     }
 
     /**
-     * Division function for REDUCE phase of AVG aggregate. Precision and scale is only used by type inference
-     * (see {@link IgniteSqlOperatorTable#DECIMAL_DIVIDE}, their values are ignored at runtime.
+     * Decimal division. Precision is only used by type inferenc, its value is ignored at runtime.
```

Comment:
```suggestion
     * Decimal division. Precision is only used by type inference, its value is ignored at runtime.
```

Reply:
why do we need precision in execution for type inference which is in preparation stage?

Reply:
This operator is used by to compute result of AVG map reduce aggregate (as it is written in the javadoc). 
We use precision/scale parameters in DIVIDE_DECIMAL (see IgniteSqlOperatorTable::DECIMAL_DIVIDE) so the result of SUM(x) / COUNT(x) returns exactly the same type as AVG(x).

Reply:
I'm fine with keeping precision in DIVIDE_DECIMAL, but why do we need to keep it in execution?

Reply:
Because RexImpTable cannot generate code a RexCall with N parameters if a java method that corresponds to this RexCall accept the number of parameters that is not equal to N.